nixos/modules/services/monitoring/netdata.nix

```diff
@@ -4,19 +4,34 @@ with lib;
 
 let
   cfg = config.services.netdata;
+  localConfig = {
+    global = {
+      "plugins directory" =  "${netdataWrapped}/libexec/netdata/plugins.d";
+    };
+  };
 
-  configFile = pkgs.writeText "netdata.conf" cfg.configText;
+  mkConfig = generators.toINI {} (recursiveUpdate localConfig cfg.config);
+  configFile = pkgs.writeText "netdata.conf" (if cfg.configText != null then cfg.configText else mkConfig);
 
   defaultUser = "netdata";
+  netdataWrapped = pkgs.stdenv.mkDerivation {
+    name = "netdataWrapped";
+    buildInputs = [ pkgs.netdata ];
+    builder = pkgs.writeText "builder.sh" ''
+      source $stdenv/setup
+      mkdir -p $out/libexec/netdata/plugins.d
+      echo '#!${pkgs.bash}/bin/bash
+      exec /run/wrappers/bin/apps.plugin $@
+      ' >$out/libexec/netdata/plugins.d/apps.plugin
+      chmod 755 $out/libexec/netdata/plugins.d/apps.plugin
```

Comment:
using `makeWrapper` for this would make it a lot nicer

Reply:
I'd agree that `makeWrapper` would be better, but I ran into some issues because `/run/wrappers/bin/`  doesn't exist when this is evaluated and it fails to build.  I think sandboxing would also cause this to fail because it's out side of the store.

This is quickly becoming moot though. https://github.com/firehol/netdata/pull/2900 should be available in the next release and I can add `/run/wrappers/bin/` to the plugin directory list and it will do the right thing without all the wrapping.